src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ExclusionZoneService.kt

```diff
@@ -0,0 +1,182 @@
+      description = description,
+      thumbnailImage = thumbnailImage,
+      uploadDetailId = savedDetail.id,
+    )
+
+    val updatedAdditionalCondition = additionalCondition.copy(additionalConditionUploadSummary = listOf(uploadSummary))
+
+    additionalConditionRepository.saveAndFlush(updatedAdditionalCondition)
+  }
+
+  @Transactional
+  fun removeExclusionZoneFile(licenceId: Long, conditionId: Long) {
+    licenceRepository
+      .findById(licenceId)
+      .orElseThrow { EntityNotFoundException("$licenceId") }
+
+    val additionalCondition = additionalConditionRepository
+      .findById(conditionId)
+      .orElseThrow { EntityNotFoundException("$conditionId") }
+
+    // Remove uploadDetail rows manually - it is intentionally not linked to the additionalCondition entity
+    additionalCondition.additionalConditionUploadSummary.map { it.uploadDetailId }.forEach {
+      additionalConditionUploadDetailRepository.findById(it).ifPresent { detail ->
+        additionalConditionUploadDetailRepository.delete(detail)
+      }
+    }
+
+    // Remove the uploadSummary via the additionalCondition
+    val updatedAdditionalCondition = additionalCondition.copy(additionalConditionUploadSummary = emptyList())
+    additionalConditionRepository.saveAndFlush(updatedAdditionalCondition)
+  }
+
+  fun getExclusionZoneImage(licenceId: Long, conditionId: Long): ByteArray? {
+    licenceRepository
+      .findById(licenceId)
+      .orElseThrow { EntityNotFoundException("$licenceId") }
+
+    val additionalCondition = additionalConditionRepository
+      .findById(conditionId)
+      .orElseThrow { EntityNotFoundException("$conditionId") }
+
+    val uploadIds = additionalCondition.additionalConditionUploadSummary.map { it.uploadDetailId }
+    if (uploadIds.isEmpty()) {
+      throw EntityNotFoundException("$conditionId")
+    }
+
+    val upload = additionalConditionUploadDetailRepository
+      .findById(uploadIds.first())
+      .orElseThrow { EntityNotFoundException("$conditionId") }
+
+    return upload.fullSizeImage
+  }
+
+  fun extractFullSizeImageJpeg(fileStream: InputStream): ByteArray? {
+    var pdfDoc: PDDocument? = null
+    try {
+      pdfDoc = PDDocument.load(fileStream)
+      val renderer = PDFRenderer(pdfDoc)
+      val firstImage = renderer.renderImage(0)
+      val baos = ByteArrayOutputStream()
+      ImageIO.write(firstImage, "jpg", baos)
+      return baos.toByteArray()
+    } catch (e: Exception) {
+      log.error("Extracting full size image - error ${e.message}")
+    } finally {
+      pdfDoc?.close()
+      fileStream.close()
+    }
+    return ByteArray(0)
+  }
+
+  fun extractDescription(fileStream: InputStream): String? {
+    var pdfDoc: PDDocument? = null
+    try {
+      pdfDoc = PDDocument.load(fileStream)
+      val stripper = PDFTextStripper()
+      stripper.sortByPosition = true
+      stripper.startPage = 2
+      return stripper.getText(pdfDoc)
+    } catch (e: Exception) {
```

Comment:
Are you able to be more specific with which type of exception might throw here? Catching general exceptions is considered bad practice - https://newbedev.com/is-it-really-that-bad-to-catch-a-general-exception#:~:text=Unless%20you%20are,everything%20is%20ok.

Reply:
Yes - as mentioned above - I can take a look at these exceptions when processing PDFs and Images. They can be more specific.

Reply:
Looks better now, thanks